### PR TITLE
Update indices-query.asciidoc

### PR DESCRIPTION
**Pull request by cbuescher**
The current description of the query seemed confusing to some people (e.g. https://twitter.com/wielinde/status/654582620630687744), this is just a proposal to simplify it.

### CODE REVIEW

**Comment by s1monw** [comment]
LGTM but I wonder if we really need that query? it's not related to this PR and you should merge as it is but I wonder what the real usecases are and if it can't be build with other tools? @clintongormley do you know what the history is here?

File: docs/reference/query-dsl/indices-query.asciidoc [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
-The `indices` query can be used when executed across multiple indices,
-allowing to have a query that executes only when executed on an index
-that matches a specific list of indices, and another query that executes
-when it is executed on an index that does not match the listed indices.
+The `indices` query is useful in cases where a search is executed across
+multiple indices. It allows to specify a list of index names and an inner
+query that is only executed for indices matching names on that list.
+For other indices that are searched, but that don't match entries
+on the list, the alternative `no_match_query` is exectuted.
```
Comment (javanna) [inline docs/reference/query-dsl/indices-query.asciidoc]: s/exectuted/executed

**Comment by javanna** [comment]
thanks for picking this up @cbuescher !

**Comment by cbuescher** [comment]
@s1monw thanks, I think I saw this query beeing used when searching an alias that covers indices with different schemas, so you need two slightly different queries for either of these. Not sure if there is a good replacement for this usecase.

**Comment by clintongormley** [comment]
We plan to remove this query in the future: https://github.com/elastic/elasticsearch/issues/12017

**Pull request merged**